doc/whats_new.rst

```diff
@@ -31,6 +31,10 @@ Changelog
 New features
 ............
 
+   - Added the :class:`sklearn.preprocessing.TransformedTargetRegressor` which
+     is a meta-estimator to regress on a modified ``y``. :issue:`9041` by
```

Comment:
just to make this more relatable, describe a small use case, maybe
"for example, to perform regression in log-space"

or something?